pkg/payload/task.go

```diff
@@ -205,5 +205,9 @@ func SummaryForReason(reason, name string) string {
 	if strings.HasPrefix(reason, "UpdatePayload") {
 		return "the update could not be applied"
 	}
-	return "an unknown error has occurred"
+
+	if len(name) > 0 {
+		return fmt.Sprintf("the cluster operator %s has an unknown error: %s", name, reason)
```

Comment:
what says the name is set for cluster operator only?

Reply:
> what says the name is set for cluster operator only?

```console
$ git --no-pager grep -B2 SummaryForReason
pkg/cvo/status.go-		if uErr, ok := err.(*payload.UpdateError); ok {
pkg/cvo/status.go-			reason = uErr.Reason
pkg/cvo/status.go:			msg = payload.SummaryForReason(reason, uErr.Name)
--
pkg/payload/task.go-}
pkg/payload/task.go-
pkg/payload/task.go:func SummaryForReason(reason, name string) string {
$ git grep -A5 UpdateError | grep Name: | grep -v _test.go
lib/resourcebuilder/apps.go-				Name:    iden,
lib/resourcebuilder/apps.go-				Name:    iden,
lib/resourcebuilder/apps.go-				Name:    iden,
lib/resourcebuilder/apps.go-				Name:    iden,
lib/resourcebuilder/apps.go-				Name:    iden,
lib/resourcebuilder/apps.go-			Name:    iden,
pkg/cvo/internal/operatorstatus.go-				Name:    expected.Name,
pkg/cvo/internal/operatorstatus.go-				Name:    actual.Name,
pkg/cvo/internal/operatorstatus.go-				Name:    actual.Name,
pkg/cvo/sync_worker.go-		Name:    name,
pkg/payload/precondition/precondition.go-		Name:    "PreconditionCheck",
$ git grep -A5 UpdateError origin/pr/185 | grep Name: | grep -v _test.go
origin/pr/185:pkg/cvo/internal/operatorstatus.go-				Name:    expected.Name,
origin/pr/185:pkg/cvo/internal/operatorstatus.go-				Name:    actual.Name,
origin/pr/185:pkg/cvo/internal/operatorstatus.go-				Name:    actual.Name,
origin/pr/185:pkg/cvo/sync_worker.go-		Name:    name,
```

So yeah, appears that I was making an assumption that is no longer true.  Dropped `the cluster operator` and rebased onto master with 27c83fc -> 2b06b79.